cartoframes/data/registry/query_dataset.py

```diff
@@ -24,12 +26,29 @@ def query(self):
 
     @property
     def dataset_info(self):
-        raise CartoException('We can not extract Dataset info from a QueryDataset. Use a TableDataset '
-                             '`Dataset(table_name)` to get or modify the info from a CARTO table.')
-
-    def update_dataset_info(self, privacy=None, name=None):
-        raise CartoException('We can not extract Dataset info from a QueryDataset. Use a TableDataset '
-                             '`Dataset(table_name)` to get or modify the info from a CARTO table.')
+        if self._dataset_info is None:
+            tables = self._get_tables_used_by_query()
+            if not tables or len(tables) == 0:
+                raise CartoException('We can not extract the table/s used by the QueryDataset, so we can not get the '
```

Comment:
there's a case we might not support with this approach: complex queries wrapped as postgres functions.

For those cases we might want to:

1. Just do not support it (add proper documentation)
2. Allow explicitly specifying a list of tables involved in the query
3. Other? 🤔

Reply:
> complex queries wrapped as postgres functions.

Can you give me an example?

Reply:
A simple one:

```
CREATE OR REPLACE FUNCTION test_01()
  RETURNS TABLE (p int) AS
$func$
BEGIN
   RETURN QUERY
   SELECT t as p
   FROM  test01;
END
```

@Jesus89 may now about this, since they wrapped a bunch of complex queries for Reveal for example, to avoid having to write them down every time. I now we have partners and clients that follow this approach as well.

Maybe not a good case for CARTOframes, but I'm not saying to support them just to take into account :)

Reply:
The key here is if `CDB_QueryTablesText` works with that or not. I am going to check it

Reply:
I have been checking CDB_QueryTablesText, and as you said, it is not getting tables from functions. For example:

```
SELECT CDB_QueryTablesText('SELECT CDB_UserTables()') as tables;
 tables 
--------
 {}
(1 row)
```

First of all, I dont any other function or other way to get tables used. And it is a special case.

So, I see the situation in a similar way as you. Forgetting the "don't support it", we could:

1. improve CDB_QueryTablesText
2. give the possibility to the user of adding the tables in this cases

About the first one, it is out of scope here. And the second, we could handle it in the https://github.com/CartoDB/cartoframes/issues/731 issue, allowing receive the list of tables used instead of looking for them with `get_table_names` method